test/metabase/db/data_source_test.clj

```diff
@@ -1,18 +1,53 @@
 (ns metabase.db.data-source-test
-  (:require [clojure.java.jdbc :as jdbc]
-            [clojure.test :refer :all]
-            [metabase.db.data-source :as mdb.data-source]
-            [metabase.test :as mt]))
+  (:require
+   [clojure.java.jdbc :as jdbc]
+   [clojure.test :refer :all]
+   [metabase.config :as config]
+   [metabase.connection-pool :as pool]
+   [metabase.db.data-source :as mdb.data-source]
+   [metabase.test :as mt]))
+
+(defn- ->DataSource [s properties]
+  (#'mdb.data-source/->DataSource s (some-> (not-empty properties) pool/map->properties)))
 
 (deftest broken-out-details-test
-  (let [data-source (mdb.data-source/broken-out-details->DataSource
-                     :h2
-                     {:subprotocol "h2"
-                      :db          (format "mem:%s" (mt/random-name))
-                      :classname   "org.h2.Driver"})]
-    (with-open [conn (.getConnection data-source)]
-      (is (= [{:one 1}]
-             (jdbc/query {:connection conn} "SELECT 1 AS one;"))))))
+  (testing :postgres
+    (is (= (->DataSource
```

Comment:
This really confused me at first until I realized you put your own notion of equality in it. But I see that you are comparing properties and the url. A problem with this is that url equality works as:

```
 Two URL objects are equal if they have the same protocol, reference equivalent hosts, have the same port number on the host, and the same file and fragment of the file.

Two hosts are considered equivalent if both host names can be resolved into the same IP addresses; else if either host name can't be resolved, the host names must be equal without regard to case; or both host names equal to null.
```
Which means we are making network requests for this comparison. That's always a weird footgun with this. I'm not sure if that's something we should live with in testing or be wary of.

Reply:
Ah. I totally misread this. Its just string comparison. Ignore me

Reply:
yup. No `URL` or `URI` involved. It's just a string and the `Properties` map which works like you'd expect